hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -167,6 +167,15 @@ object Simplify {
     case ApplyIR("indexArray", Seq(a, i@I32(v))) if v >= 0 =>
       ArrayRef(a, i)
 
+    case ToArray(x) if x.typ.isInstanceOf[TArray] => x
+
+    case ApplyIR("contains", Seq(ToArray(x), element)) if x.typ.isInstanceOf[TSet] => invoke("contains", x, element)
+
+    case ApplyIR("contains", Seq(Literal(t, v), element)) if t.isInstanceOf[TArray] =>
+      invoke("contains", Literal(TSet(t.asInstanceOf[TArray].elementType, t.required), v.asInstanceOf[IndexedSeq[_]].toSet), element)
+
+    case ApplyIR("contains", Seq(ToSet(x), element)) if x.typ.isInstanceOf[TArray] => invoke("contains", x, element)
```

Comment:
What's the rationale for this change? It seems to me it would be slower.

Reply:
Array contains requires N comparisons, while `ToSet(array) contains` will do N log N to create the set (with a full copy!), and then another logN to binary search for the element.

If someone is going to create a set to use with `.contains` and then immediately discard it, we can save that work!

Reply:
ok. that's what i figured but wanted to make sure.